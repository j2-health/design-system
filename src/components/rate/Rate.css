.j2-rate.ant-rate {
  color: var(--j2-gold-6);
}
