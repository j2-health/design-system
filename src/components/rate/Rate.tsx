import { StarIcon } from '@phosphor-icons/react'
import { Rate as AntdRate, RateProps } from 'antd'
import './Rate.css'

export type Props = Expand<RateProps>

export const Rate = ({ count = 5, ...props }: Props) => {
  return (
    <AntdRate
      {...props}
      className="j2-rate"
      count={count}
      disabled={true}
      allowHalf={true}
      allowClear={true}
      character={<StarIcon weight="fill" />}
    />
  )
}
