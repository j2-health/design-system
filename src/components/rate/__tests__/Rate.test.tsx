import { render } from '@testing-library/react'
import { Rate } from '../Rate'

describe('Rate', () => {
  it('should render correctly', () => {
    const { container } = render(<Rate value={4.5} />)
    expect(container).toMatchSnapshot()
  })
})
