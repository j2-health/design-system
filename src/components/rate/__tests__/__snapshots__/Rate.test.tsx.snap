// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rate should render correctly 1`] = `
<div>
  <ul
    class="ant-rate j2-rate css-dev-only-do-not-override-os2fvf ant-rate-disabled"
    tabindex="-1"
  >
    <li
      class="ant-rate-star ant-rate-star-full"
    >
      <div
        aria-checked="true"
        aria-posinset="1"
        aria-setsize="5"
        role="radio"
        tabindex="-1"
      >
        <div
          class="ant-rate-star-first"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M234.29,114.85l-45,38.83L203,211.75a16.4,16.4,0,0,1-24.5,17.82L128,198.49,77.47,229.57A16.4,16.4,0,0,1,53,211.75l13.76-58.07-45-38.83A16.46,16.46,0,0,1,31.08,86l59-4.76,22.76-55.08a16.36,16.36,0,0,1,30.27,0l22.75,55.08,59,4.76a16.46,16.46,0,0,1,9.37,28.86Z"
            />
          </svg>
        </div>
        <div
          class="ant-rate-star-second"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M234.29,114.85l-45,38.83L203,211.75a16.4,16.4,0,0,1-24.5,17.82L128,198.49,77.47,229.57A16.4,16.4,0,0,1,53,211.75l13.76-58.07-45-38.83A16.46,16.46,0,0,1,31.08,86l59-4.76,22.76-55.08a16.36,16.36,0,0,1,30.27,0l22.75,55.08,59,4.76a16.46,16.46,0,0,1,9.37,28.86Z"
            />
          </svg>
        </div>
      </div>
    </li>
    <li
      class="ant-rate-star ant-rate-star-full"
    >
      <div
        aria-checked="true"
        aria-posinset="2"
        aria-setsize="5"
        role="radio"
        tabindex="-1"
      >
        <div
          class="ant-rate-star-first"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M234.29,114.85l-45,38.83L203,211.75a16.4,16.4,0,0,1-24.5,17.82L128,198.49,77.47,229.57A16.4,16.4,0,0,1,53,211.75l13.76-58.07-45-38.83A16.46,16.46,0,0,1,31.08,86l59-4.76,22.76-55.08a16.36,16.36,0,0,1,30.27,0l22.75,55.08,59,4.76a16.46,16.46,0,0,1,9.37,28.86Z"
            />
          </svg>
        </div>
        <div
          class="ant-rate-star-second"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M234.29,114.85l-45,38.83L203,211.75a16.4,16.4,0,0,1-24.5,17.82L128,198.49,77.47,229.57A16.4,16.4,0,0,1,53,211.75l13.76-58.07-45-38.83A16.46,16.46,0,0,1,31.08,86l59-4.76,22.76-55.08a16.36,16.36,0,0,1,30.27,0l22.75,55.08,59,4.76a16.46,16.46,0,0,1,9.37,28.86Z"
            />
          </svg>
        </div>
      </div>
    </li>
    <li
      class="ant-rate-star ant-rate-star-full"
    >
      <div
        aria-checked="true"
        aria-posinset="3"
        aria-setsize="5"
        role="radio"
        tabindex="-1"
      >
        <div
          class="ant-rate-star-first"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M234.29,114.85l-45,38.83L203,211.75a16.4,16.4,0,0,1-24.5,17.82L128,198.49,77.47,229.57A16.4,16.4,0,0,1,53,211.75l13.76-58.07-45-38.83A16.46,16.46,0,0,1,31.08,86l59-4.76,22.76-55.08a16.36,16.36,0,0,1,30.27,0l22.75,55.08,59,4.76a16.46,16.46,0,0,1,9.37,28.86Z"
            />
          </svg>
        </div>
        <div
          class="ant-rate-star-second"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M234.29,114.85l-45,38.83L203,211.75a16.4,16.4,0,0,1-24.5,17.82L128,198.49,77.47,229.57A16.4,16.4,0,0,1,53,211.75l13.76-58.07-45-38.83A16.46,16.46,0,0,1,31.08,86l59-4.76,22.76-55.08a16.36,16.36,0,0,1,30.27,0l22.75,55.08,59,4.76a16.46,16.46,0,0,1,9.37,28.86Z"
            />
          </svg>
        </div>
      </div>
    </li>
    <li
      class="ant-rate-star ant-rate-star-full"
    >
      <div
        aria-checked="true"
        aria-posinset="4"
        aria-setsize="5"
        role="radio"
        tabindex="-1"
      >
        <div
          class="ant-rate-star-first"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M234.29,114.85l-45,38.83L203,211.75a16.4,16.4,0,0,1-24.5,17.82L128,198.49,77.47,229.57A16.4,16.4,0,0,1,53,211.75l13.76-58.07-45-38.83A16.46,16.46,0,0,1,31.08,86l59-4.76,22.76-55.08a16.36,16.36,0,0,1,30.27,0l22.75,55.08,59,4.76a16.46,16.46,0,0,1,9.37,28.86Z"
            />
          </svg>
        </div>
        <div
          class="ant-rate-star-second"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M234.29,114.85l-45,38.83L203,211.75a16.4,16.4,0,0,1-24.5,17.82L128,198.49,77.47,229.57A16.4,16.4,0,0,1,53,211.75l13.76-58.07-45-38.83A16.46,16.46,0,0,1,31.08,86l59-4.76,22.76-55.08a16.36,16.36,0,0,1,30.27,0l22.75,55.08,59,4.76a16.46,16.46,0,0,1,9.37,28.86Z"
            />
          </svg>
        </div>
      </div>
    </li>
    <li
      class="ant-rate-star ant-rate-star-half ant-rate-star-active"
    >
      <div
        aria-checked="true"
        aria-posinset="5"
        aria-setsize="5"
        role="radio"
        tabindex="-1"
      >
        <div
          class="ant-rate-star-first"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M234.29,114.85l-45,38.83L203,211.75a16.4,16.4,0,0,1-24.5,17.82L128,198.49,77.47,229.57A16.4,16.4,0,0,1,53,211.75l13.76-58.07-45-38.83A16.46,16.46,0,0,1,31.08,86l59-4.76,22.76-55.08a16.36,16.36,0,0,1,30.27,0l22.75,55.08,59,4.76a16.46,16.46,0,0,1,9.37,28.86Z"
            />
          </svg>
        </div>
        <div
          class="ant-rate-star-second"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M234.29,114.85l-45,38.83L203,211.75a16.4,16.4,0,0,1-24.5,17.82L128,198.49,77.47,229.57A16.4,16.4,0,0,1,53,211.75l13.76-58.07-45-38.83A16.46,16.46,0,0,1,31.08,86l59-4.76,22.76-55.08a16.36,16.36,0,0,1,30.27,0l22.75,55.08,59,4.76a16.46,16.46,0,0,1,9.37,28.86Z"
            />
          </svg>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
